planner/core/physical_plans.go

```diff
@@ -186,6 +186,20 @@ func (p *PhysicalTableReader) GetAvgRowSize() float64 {
 	return getTblStats(p.tablePlan).GetAvgRowSize(p.ctx, p.tablePlan.Schema().Columns, false, false)
 }
 
+// MemoryUsage return the memory usage of PhysicalTableReader
+func (p *PhysicalTableReader) MemoryUsage() (sum int64) {
+	if p == nil {
+		return
+	}
+
+	sum = p.physicalSchemaProducer.MemoryUsage() + size.SizeOfUint8*2 + size.SizeOfBool + p.PartitionInfo.MemoryUsage()
+	// todo: memtrace: 	p.TablePlans p.PhysicalPlan
```

Comment:
Why not add the function `MemoryUsage() (sum int64)` to the `PhysicalPlan` interface? 
Then we can accumulate their memory usage like `sum += p.tablePlan.MemoryUsage()`.